experimental/framework/tinylicious-client/src/test/TinyliciousClient.spec.ts

```diff
@@ -5,42 +5,364 @@
+            return true;
+        };
+
+        await assert.doesNotReject(
+            containerAndServices,
+            errorFn,
+            "container cannot be created with port number",
+        );
+    });
+
+    /**
+    * Scenario: test when TinyliciousClient is initialized correctly, it can create
+    * a container successfully.
+    *
+    * Expected behavior: an error should not be thrown nor should a rejected promise
+    * be returned.
+    */
+    it("can create a container and services successfully", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const containerAndServices = ResettableTinyliciousClient.createContainer(containerConfig, schema);
+        const errorFn = (error) => {
+            assert.notStrictEqual(error.message, undefined, "Container and services cannot be created");
+            return true;
+        };
+
+        await assert.doesNotReject(
+            containerAndServices,
+            errorFn,
+            "TinyliciousClient cannot create container and services successfully",
+        );
+    });
+
+    /**
+    * Scenario: Given the container already exists, test that TinyliciousClient can get the existing container
+    * when provided with valid ContainerConfig and ContainerSchema.
+    *
+    * Expected behavior: containerCreate should have the identical SharedMap ID as containerGet.
+    */
+    it("can get a container successfully", async () => {
+        const containerConfig: TinyliciousContainerConfig = { id: documentId };
+        const schema: ContainerSchema = {
+            name: documentId,
+            initialObjects: {
+                map1: SharedMap,
+            },
+        };
+        const [containerCreate] = await ResettableTinyliciousClient.createContainer(containerConfig, schema);
+        await new Promise<void>((resolve, reject) => {
+            containerCreate.on("connected", () => {
+                resolve();
+            });
+        });
+        const [containerGet] = await ResettableTinyliciousClient.getContainer(containerConfig, schema);
+        const map1Create = containerCreate.initialObjects.map1 as SharedMap;
+        const map1Get = containerGet.initialObjects.map1 as SharedMap;
+        assert.strictEqual(map1Get.id, map1Create.id, "Error getting a container");
+    });
+
+    /**
+    * Scenario: test if an initialized TinyliciousClient can create container successfully when
+    * it is already initialized.
+    *
+    * Expected behavior: TinyliciousClient should use the existing instance instead of creating
+    * a new instance, then proceed to create a container.
+    */
+    it("can initialize an already initialized TinyliciousClient", async () => {
+        ResettableTinyliciousClient.init(clientConfig);
```

Comment:
Is `clientConfig` needed for this test? If not, we can just define it in the one test that is using it

Reply:
haha nope. good catch. I removed it